### PR TITLE
Bugfix for custom prefix/postfix and metric collection

### LINKED ISSUE #2065: Prefix clash with MetricCollection with 1.1.1
## 🐛 Bug

There is a bug that happens with metrics that returns dictionaries when used in MetricCollection.

### To Reproduce

Steps to reproduce the behavior...

```python
import torch
from torchmetrics import Metric, MetricCollection

class CustomAccuracy(Metric):
    def update(self, preds: torch.Tensor, target: torch.Tensor) -> None:
        self.correct = torch.sum(preds == target)
        self.total = preds.numel()
        self.prefix = 'accuracy'

    def compute(self) -> torch.Tensor:
        res = self.correct.float() / self.total
        return {f"{self.prefix}/value": res}

class CustomPrecision(Metric):
    def update(self, preds: torch.Tensor, target: torch.Tensor) -> None:
        self.true_positives = torch.sum((preds == target) & (preds == 1))
        self.predicted_positives = torch.sum(preds == 1)
        self.prefix = 'precision'

    def compute(self) -> torch.Tensor:
        res =  self.true_positives.float() / self.predicted_positives
        return {f"{self.prefix}/value": res}

# Initialize MetricCollection with Accuracy and Precision
metrics = MetricCollection([CustomAccuracy(), CustomPrecision()])

# Mock predictions and targets
preds = torch.tensor([1, 0, 0, 1])
targets = torch.tensor([1, 0, 0, 0])

# Update metrics with current batch
metrics(preds, targets)

# Print the calculated metrics
print(metrics.compute())
# Returns {'precisionaccuracy/value': tensor(0.7500), 'precisionprecision/value': tensor(0.5000)}
```

<details>
  <summary>Code sample</summary>

</details>

### Expected behavior

I expect the results to be 

`{'accuracy/value': tensor(0.7500), 'precision/value': tensor(0.5000)}`

and not 

`{'precisionaccuracy/value': tensor(0.7500), 'precisionprecision/value': tensor(0.5000)}`

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): 1.1.1, via pip
- Python & PyTorch Version: 3.10 and 1.13
- Any other relevant information such as OS (e.g., Linux):

### Additional context

Please note that this works correctly in torchmetrics 0.11.4

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2065
Make sure that metric collection does not conflict with user attributes `prefix`/`postfix`

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2070.org.readthedocs.build/en/2070/

### CODE REVIEW

File: src/torchmetrics/collections.py [hunk 1]
```diff
@@ -339,17 +339,18 @@ def _compute_and_reduce(
...
+            res = result[k]
             if isinstance(res, dict):
                 for key, v in res.items():
                     # if duplicates of keys we need to add unique prefix to each key
                     if duplicates:
                         stripped_k = k.replace(getattr(m, "prefix", ""), "")
                         stripped_k = stripped_k.replace(getattr(m, "postfix", ""), "")
                         key = f"{stripped_k}_{key}"
-                    if hasattr(m, "prefix") and m.prefix is not None:
+                    if hasattr(m, "_from_collection") and m._from_collection and m.prefix is not None:
```
Comment (Borda) [inline src/torchmetrics/collections.py:351]:
```suggestion
                    if getattr(m, "_from_collection") and m.prefix is not None:
```

**Review by Borda — APPROVED** [review]

**Review by stancld — APPROVED** [review]
Looks good, thanks!

**Review by justusschock — APPROVED** [review]
